mypy/plugins/dataclasses.py

```diff
@@ -106,22 +105,13 @@ def to_argument(self, current_info: TypeInfo) -> Argument:
             arg_kind = ARG_OPT
         return Argument(
             variable=self.to_var(current_info),
-            type_annotation=self.expand_type(current_info),
+            type_annotation=expand_self_type(self.type, self.info, fill_typevars(current_info)),
             initializer=None,
             kind=arg_kind,
         )
 
-    def expand_type(self, current_info: TypeInfo) -> Optional[Type]:
-        if self.type is not None and self.info.self_type is not None:
-            # In general, it is not safe to call `expand_type()` during semantic analyzis,
-            # however this plugin is called very late, so all types should be fully ready.
-            # Also, it is tricky to avoid eager expansion of Self types here (e.g. because
-            # we serialize attributes).
-            return expand_type(self.type, {self.info.self_type.id: fill_typevars(current_info)})
-        return self.type
-
     def to_var(self, current_info: TypeInfo) -> Var:
-        return Var(self.name, self.expand_type(current_info))
+        return Var(self.name, expand_self_type(self.type, self.info, fill_typevars(current_info)))
```

Comment:
I think that it still makes sense to keep `expand_type` as a method that just calls `expand_self_type`. This way you can preserve the original comment in `expand_type` about the safety of calling `expand_self_type` (which is similar to calling `expand_type`).

Reply:
I found `expand_self_type` while refactoring out `expand_type` that was being used by `dataclasses`, as I happen to have named them the same way and my IDE picked up the wrong one when attempting to import it.

How would naming work here if they both act the same way and basically have the same code? Should I just keep them both as `expand_self_type` and just add the note in the one in `common.py`?